core/block.js

```diff
@@ -1711,6 +1721,9 @@ Blockly.Block.prototype.moveNumberedInputBefore = function(
  */
 Blockly.Block.prototype.removeInput = function(name, opt_quiet) {
   for (var i = 0, input; (input = this.inputList[i]); i++) {
+    if (input.type == Blockly.NEXT_STATEMENT) {
+      this.statementInputs--;
+    }
     if (input.name == name) {
       input.dispose();
       this.inputList.splice(i, 1);
```

Comment:
Should we only decrease statementInputs count when we dispose of an input?

Reply:
Whoops, yes! Good catch